docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
-    - command: kustomize build .
-  updaters:
-    # use https://github.com/squaremo/kubeyaml on flux-patch.yaml
-    - containerImage:
-        command: >-
-          cat flux-patch.yaml |
-          kubeyaml image --namespace $FLUX_WL_NS --kind $FLUX_WL_KIND --name $FLUX_WL_NAME --container $FLUX_CONTAINER --image "$FLUX_IMG:$FLUX_TAG"
-          > new-flux-patch.yaml &&
-          mv new-flux-patch.yaml flux-patch.yaml
-      policy:
-        command: >-
-          cat flux-patch.yaml |
-          kubeyaml annotate --namespace $FLUX_WL_NS --kind $FLUX_WL_KIND --name $FLUX_WL_NAME "fluxcd.io/$FLUX_POLICY=$FLUX_POLICY_VALUE"
-          > new-flux-patch.yaml &&
-          mv new-flux-patch.yaml flux-patch.yaml
+  - command: kustomize build .
+  patchFile: flux-patch.yaml
 ```
 
-For every flux target path, Flux will look for a `.flux.yaml` file in the target path and all its parent directories.
-If a `.flux.yaml` is found:
+The `generators` field is an array of commands, all of which will be
+run in turn. Each command is expected to print a YAML stream to its
+stdout. The streams are concatenated and parsed as one big YAML
+stream, before being applied.
 
-1. When syncing, `fluxd` will run each of the `generators`, collecting yaml manifests printed to `stdout` and applying
-   them to the cluster.
-2. When making a release or updating a policy, `fluxd` will run the `updaters`, which are in charge of updating the Git
-   repository to reflect the required changes in workloads.
-    * The `containerImage` updaters are invoked once for every container whose image requires updating.
-    * The `policy` updaters are invoked once for every workload annotation which needs to be added or updated.
-    * Updaters are supplied with environment variables indicating what image should be updated and what annotation to
-      update (more on this later).
-    * Updaters expected to modify the Git working tree in-place
+Much of the time, it will only be necessary to supply one command to
+be run.
 
-    After invoking the updaters, `fluxd` will then commit and push the resulting modifications to the git repository.
+The commands will be run with the `--git-path` being processed as a
+working directory -- which is not necessarily the same directory in
+which the `.flux.yaml` file was found. [See below](#execution-context)
+for more details on the execution context in which commands are run.
 
-3. `fluxd` will ignore any other yaml files under that path (e.g. resource manifests).
+### Using patch-updated configuration
 
-Generators and updaters are intentionally independent, in case a matching updater cannot be provided. It is hard to
-create updaters for some factorization technologies (particularly Configuration-As-Code). To improve the situation,
-a separate configuration file variant (`patchedUpdated`) is provided, which will be described later on.
+A patch-updated configuration generates manifests using commands, and
+records updates in a patch file.
```

Comment:
Is it worth mentioning it's a Strategic Merge patch? We could hide that fact to avoid confusion, but it's their responsibility consistent with the manifests they generate.